Source/multi.cpp

```diff
@@ -311,10 +312,10 @@ void ProcessTmsgs()
 
 void SendPlayerInfo(int pnum, _cmd_id cmd)
 {
-	PkPlayerStruct pkplr;
+	std::unique_ptr<byte[]> pkplr { new byte[sizeof(PkPlayerStruct)] };

```

Comment:
This needs to be aligned to `PkPlayerStruct` alignment

Reply:
Already is. alignof(PkPlayerStruct) == 1

Reply:
Perhaps add a `static_assert` for that? As it is, the code is not robust to alignment changes in `PkPlayerStruct`.

Reply:
Thanks! Looking at the code, it appears that its alignment is ensured by `#pragma pack`, which may not be supported on some platforms. It'd be nice if we didn't rely on that but it's not a big deal with the static assert in place.

Reply:
I really don't think pragma pack or attribute(packed) lack support from any serious compiler. Network code, especially TCP/IP stack implementations, rely on this to force the layout of certain structs.